metasynth/distribution/regex.py

```diff
@@ -0,0 +1,94 @@
+"""Distribution for structured strings, using regexes."""
+from __future__ import annotations
+
+from typing import Union, Optional
+
+from regexmodel import RegexModel, NotFittedError
+
+from metasynth.distribution.base import metadist, BaseDistribution, UniqueDistributionMixin
+
+
+@metadist(implements="core.regex", var_type="string", version="2.0")
+class RegexDistribution(BaseDistribution):
+    """Distribution for structured strings.
+
+    Examples that this distribution should work reasonably for are: email,
+    ID's, telephone numbers, ip addresses, etc.
+
+    Valid regexes are ([a][b]|[c][d]), or [A-Z][0-9]{2,3}. During the fit, the probabilities
+    of each of the branches in the regex is also recorded so that synthetic values can be
+    drawn from a better distribution.
+
+    Parameters
+    ----------
+    regex_data:
+        A valid input for the regex model. The two main ones is a serialized version
+        of the model, and the other is a regex that satisfies falls within the language
+        of the regexmodel package (which is a small subset of the python re package).
+    """
```

Comment:
This should be improved, make it clear there are two options and potentially refer advanced users to the regexmodel package